### PR TITLE
net-nds/openldap: update to 2.5.14

### PR DESCRIPTION
**Pull request by dongsupark**
Update `net-nds/openldap` to 2.5.14 mainly to address [CVE-2023-2953](https://nvd.nist.gov/vuln/detail/CVE-2023-2953).

Based on Gentoo commit [5b1520ea69d4](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=5b1520ea69d410ab743eb7998b5d886bbf597375).

mask openldap 2.6.3

Mask openldap 2.6.3+ in overlay profiles, to take different steps of updating openldap.

1) from 2.4 to 2.5,
2) do an Alpha release around 2023-08, and
3) finally update from 2.5 to 2.6.

Fixes https://github.com/flatcar/Flatcar/issues/1120.

## Testing done

CI: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/2191/cldsv/

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

**Comment by dongsupark** [comment]
CI passed

File: changelog/updates/2023-07-20-openldap-2.5.14.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- openldap ([2.5.14](https://lists.openldap.org/hyperkitty/list/openldap-announce@openldap.org/thread/TZQHR4SIWUA5BZTKDAKSFDOOGDVU4TU7/))
```
Comment (krnowak) [inline changelog/updates/2023-07-20-openldap-2.5.14.md:1]: Would it be a good idea to point to the 2.5 major version release announcement too?

```suggestion
- openldap ([2.5.14](https://lists.openldap.org/hyperkitty/list/openldap-announce@openldap.org/thread/TZQHR4SIWUA5BZTKDAKSFDOOGDVU4TU7/) (includes [2.5.4](https://lists.openldap.org/hyperkitty/list/openldap-announce@openldap.org/thread/BH3VDPG6IYYF5L5U6LZGHHKMJY5HFA3L/)))
```

File: sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/package.mask [hunk 2]
```diff
@@ -20,3 +20,8 @@
 # Python 3.12 is in portage-stable (currently testing), so avoid picking it
 # up. Update this to mask later versions when we switch to 3.11.
 >=dev-lang/python-3.12
+
+# Do not update to openldap 2.6.3+, to take 2 different steps of updating
+# openldap, 1) from 2.4 to 2.5, 2) do an Alpha release around 2023-08, and
+# 3) finally update from 2.5 to 2.6.
+>=net-nds/openldap-2.6.3
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/package.mask:27]: Normally I'd suggest the following, but I don't think Gentoo is going to reintroduce 2.6.1.

```suggestion
>=net-nds/openldap-2.6
```

**Review by krnowak — APPROVED** [review]

File: sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/package.mask (hunk 2 above)
Comment (dongsupark) [reply sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/package.mask]: Done

File: changelog/updates/2023-07-20-openldap-2.5.14.md (hunk 1 above)
Comment (krnowak) [reply changelog/updates/2023-07-20-openldap-2.5.14.md]: Either one is fine, 2.5.4 was the first version from 2.5 to be released for general use.

**Pull request merged**